### ISSUE TITLE
form.forceValidate() doesn't work

### ISSUE THREAD

**Issue by Oopscurity** [theadam/react-inform#12]
Hello,
Let's consider the situation when we have `handleSubmit()` like this:
```javascript
handleSubmit = (e) => {
  const { form } = this.props;

  form.forceValidate(); // (doesn't validate!)
  if (!form.isValid()) {
    return;
  }

  // submitting...

  form.onValues({}); // clean the form
  form.forceValidate(); // (also doesn't validate)
}
```
The problem is that `form.forceValidate()` at the end of the function doesn't work: validation function won't be called. So, the form "is still valid" - you can try to submit again... All will be ok, `forceValidate()` at the beginning of the function also doesn't work -> `form.isValid()` returns `true`. The form'll be validated only after *real* change like passing a character to its field. `form.onValues()` doesn't cause validation.

What's the purpose of `form.forceValidate()`, if it doesn't validate the form?
Why doesn't `form.onValues()` cause the validation, like common form changes?

**Comment by theadam** [comment]
`forceValidate` is perhaps incorrectly named.  The form should always be keeping an up to date internal validation state, but that state is only passed to the UI after the fields are "touched" (technically onblur).  This functionality is working as designed.

The issue, however, seems to be that onValues is not updating the form's values in a way that would update the internal error state of the fields.  `onValues` should be using the form's setValues method, but its not, its calling setState directly.

**Comment by theadam** [comment]
fixed by #13
